hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -169,9 +170,14 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     TableOperations ops = newTableOps(identifier);
     TableMetadata lastMetadata;
-    if (purge && ops.current() != null) {
-      lastMetadata = ops.current();
-    } else {
+    try {
+      if (purge && ops.current() != null) {
+        lastMetadata = ops.current();
+      } else {
+        lastMetadata = null;
```

Comment:
Style nit: Wondering, while we are in this code, it may be a bit more concise to initialize lastMetadata = null, then we can remove two null assignments in the else/catch blocks?  Not sure what people think, though its not a big in any case.

Reply:
Might be a personal preference, but I also prefer initialising to null and then save an else branch of an if.

Reply:
It is better to initialize to null from the concise code, but I think it is easier to understand in terms of readability.  
I think It depends on whether the `lastMetadata = null` has a significant effect on the method . If it is not that important, we can initialise it as null.

Reply:
`lastMetadata = null` seems not that important for `dropTable`. So I initialise it as null.